### DISCUSSION TITLE
Instantiating from the assets

### DISCUSSION THREAD

**Discussion by evanAndersonDigipen** [nesrak1/AssetsTools.NET#63] [Q&A]
How would i cast a gameobject from the assets to an actual gameobject to instantiate it?

**Comment by nesrak1** [comment]
Doing exactly what you suggest is not possible. I should ask what you're trying to do and if this is from a bundle or assets file.

**Reply by evanAndersonDigipen** [reply]
From an assets file, is there a workaround that you know of? I probably worded this wrong, is there any way to just instantiate a loaded asset is basically what im asking

**Reply by nesrak1** [reply]
There's not really a _workaround_ as this is quite a complex process. The main issue with GameObjects is trying to resolve dependencies and fix them to load correctly in the editor. https://github.com/nesrak1/HKWorldEdit2 or https://github.com/nesrak1/HKWorldEdit are example of loading a scene into the editor by importing the assets into a unity .scene file or an asset bundle, but it requires recursively iterating through all dependencies and fixing path ids, texture paths, shader settings, etc.

TLDR: there is no built in way into the library to do this at the moment.

**Reply by evanAndersonDigipen** [reply]
Is there a specific script  that this is done in in the given repos?

**Reply by nesrak1** [reply]
It's definitely not copy and paste into your own project, but if you want to look at the code that generates the editor files, see https://github.com/nesrak1/HKWorldEdit2/blob/master/Assets/Editor/Bundler/Loader.cs or https://github.com/nesrak1/HKWorldEdit2/blob/master/Assets/Editor/Bundler/ReferenceCrawlerEditor.cs. I don't provide any support with this right now.
